lcls-twincat-pmps/PMPS/MajorComponents/FB_TempSensor_FFO.TcPOU

```diff
@@ -0,0 +1,88 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_TempSensor_FFO" Id="{e7283f0f-d3c0-4044-8021-5daf2e8ba234}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_TempSensor_FFO EXTENDS FB_TempSensor
+VAR_INPUT
+	{attribute 'pytmc' :=
+        pv: FAULT_SP
+        io: input
+        field: EGU C
+        field: PREC 2
+    '}
+	fFaultThreshold : LREAL; //Faults when the threshold is reached. Trigger value.
+	{attribute 'pytmc' :=
+        pv: FAULT_SP_HYS
+        io: input
+        field: EGU %
+        field: PREC 2
+    '}
+	fHysteresis : LREAL :=1; // percentage determining how far below the trigger value the fault should be released
+	sDevName: STRING;
+	bVeto : BOOL:=FALSE; // This Fault can be Vetoed if the bVeto is TRUE
+END_VAR
+VAR_IN_OUT
+	io_fbFFHWO    :    FB_HardwareFFOutput;
+END_VAR
+VAR_OUTPUT
+END_VAR
+VAR
+	{attribute 'instance-path'}
+	{attribute 'noinit'}
+	sPath: STRING;
+	bFAULT_OK:BOOL :=FALSE;
+	FFO    :    FB_FastFault :=(
+        i_Desc := 'Fault occurs when the temprature trip point is reached',
+        i_TypeCode := 16#f700);
+	rtRESET	:	R_TRIG;
+	ftFAULT	:	F_TRIG;
+	ftConnected	:	F_TRIG;
+	fbLogger : FB_LogMessage := (eSubsystem:=E_SubSystem.MPS, nMinTimeViolationAcceptable:=10);
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[ // The manual states that we are disconnected if we are both overrange and in an error state
+bConnected := NOT (bOverrange AND bError);
+fTemp := INT_TO_LREAL(iRaw) * fResolution;
+
+//Verify Hysteresis is between 1-100, Shouldn't be 0. shouldn't be a 100 either.
+fHysteresis:=  LIMIT(1, fHysteresis, 100);
```

Comment:
I think 1 is a good choice for a lower limit. Imagine a typical threshold of like 50C -> 1% is 49.5C